src/primitives/transaction.h

```diff
@@ -105,7 +136,69 @@ class CTxIn
 
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action) {
-        READWRITE(prevout);
+
+        //
+        // ELEMENTS:
+
+        //TODO(rebase) CA/CT
+        //bool fHasAssetIssuance;
+        COutPoint outpoint;
+        if (!ser_action.ForRead()) {
+            //if (prevout.n == (uint32_t) -1) {
+            //    // Coinbase inputs do not have asset issuances attached
+            //    // to them.
+            //    fHasAssetIssuance = false;
+            //    outpoint = prevout;
+            //} else {
+            //    // The issuance and pegin bits can't be set as it is used to indicate
+            //    // the presence of the asset issuance or pegin objects. They should
+            //    // never be set anyway as that would require a parent
+            //    // transaction with over one billion outputs.
+            //    assert(!(prevout.n & ~COutPoint::OUTPOINT_INDEX_MASK));
+            //    // The assetIssuance object is used to represent both new
+            //    // asset generation and reissuance of existing asset types.
+            //    fHasAssetIssuance = !assetIssuance.IsNull();
+                // The mode is placed in the upper bits of the outpoint's
+                // index field. The IssuanceMode enum values are chosen to
+                // make this as simple as a bitwise-OR.
+                outpoint.hash = prevout.hash;
+                outpoint.n = prevout.n & COutPoint::OUTPOINT_INDEX_MASK;
```

Comment:
This is your genesis block problem. A coinbase's outpoint has `n==0xffffffff`, which is being masked off here.

Reply:
YES! Thanks! I was looking at this line, but somehow assumed the coinbase outpoint was 0..